marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/BuildConstraintsSpec.hs

```diff
@@ -223,3 +236,88 @@ withdrawSpec = Hspec.describe "buildWithdrawConstraints" do
           }
 
     pure $ actual `shouldBe` expected
+
+buildApplyInputsConstraintsSpec :: Spec
+buildApplyInputsConstraintsSpec =
+  Hspec.describe "buildApplyInputsConstraints" do
+    let
+      -- System start and era history are reusable across tests.
+      systemStart = SystemStart $ posixSecondsToUTCTime 0  -- Without loss of generality.
+      eraParams = EraParams
+        { eraEpochSize = 1
+        , eraSlotLength = mkSlotLength 1
+        , eraSafeZone = UnsafeIndefiniteSafeZone
+        }
+      oneSecondBound i = Bound
+        { boundTime = RelativeTime $ fromInteger i
+        , boundSlot = fromInteger i
+        , boundEpoch = fromInteger i
+        }
+      oneSecondEraSummary i = EraSummary
+        { eraStart = oneSecondBound i
+        , eraEnd = EraEnd $ oneSecondBound $ i + 1
+        , eraParams
+        }
+      unboundedEraSummary i = EraSummary
+        { eraStart = oneSecondBound i
+        , eraEnd = EraUnbounded
+        , eraParams
+        }
+      eraHistory =
+        EraHistory CardanoMode
+          . mkInterpreter
+          . summaryWithExactly
+          $ Exactly
+          $  K (oneSecondEraSummary 0) -- Byron lasted 1 second
+          :* K (oneSecondEraSummary 1) -- Shelley lasted 1 second
+          :* K (oneSecondEraSummary 2) -- Allegra lasted 1 second
+          :* K (oneSecondEraSummary 3) -- Mary lasted 1 second
+          :* K (oneSecondEraSummary 4) -- Alonzo lasted 1 second
+          :* K (unboundedEraSummary 5) -- Babbage never ends
+          :* Nil
```

Comment:
We need to move this to a helper module at some point

Reply:
Yes, and we don't even have Jira tickets yet for the comprehensive unit tests on this function.

Reply:
https://input-output.atlassian.net/browse/SCP-5042